src/operator/convolution.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+import edgeHandlingMethod from './edgeHandlingMethod'
+
+//convolution using a square kernel
+export default function convolution(newImage, kernel, edgeHandling){
+    let kernelWidth = Math.sqrt(kernel.length);
+    let k = Math.floor(kernelWidth/2);
+    let div = 0;
+    edgeHandling = edgeHandling || 'copy';
```

Comment:
See comment above for the definition of default values in the "options" parameter.